src/linear_algebra/free_module.lean

```diff
@@ -0,0 +1,350 @@
+/-
+Copyright (c) 2020 Anne Baanen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Anne Baanen
+-/
+import linear_algebra.basis
+import linear_algebra.finsupp_vector_space
+import ring_theory.principal_ideal_domain
+
+/-! # Free modules
+
+A free `R`-module `M` is a module with a basis over `R`,
+equivalently it is an `R`-module linearly equivalent to `ι →₀ R` for some `ι`.
```

Comment:
Is it worth adding `abbreviation free_module := ι →₀ R` to this file, just so that people using doc search find it?

Reply:
I don't think this file actually proves anything for `ι →₀ R` itself, only indirectly via `is_basis`. (Spoiler alert: I'm even experimenting with [redefining a basis](https://github.com/leanprover-community/mathlib/blob/bundled-basis/src/linear_algebra/basis.lean#L77) to be the linear equivalence with `ι →₀ R`.)